### ISSUE TITLE
Loading custom plugins in emqttd cluster

### ISSUE THREAD

**Issue by abhidtu** [emqx/emqx#423]
Hi @erylee this is more of a discussion,
Im working on this project https://github.com/abhidtu/emqttd_publish_client_activity
This plugins works using hooks on various emqttd events to publish client activity to mqtt or kafka on SYSTEM/# channels. 

When i'm loading this plugin on all the nodes in emqttd cluster I have this doubt in mind that will these event hooks  - on_client_connected, on_client_disconnected, on_client_subscribe, on_message_publish, on_message_acked etc always get triggered only on one node among the cluster or simultaneously many.

For example
1> in a cluster of two nodes A & B, if I publish a message to to A then is it guaranteed that the on_message_publish event hook from node A will only be triggered, is there any chance that both nodes A & B may trigger on_message_publish hook in their individual plugins (inconsistency).

2>Similarly, if a client connected to node B acknowledges a message then will the event hook "on_message_acked" will only be called on node B and not A.

In other words is there any chance of inconsistency here.

I checked this myself, same event hooks were not getting triggered but everything was distributed like on_message_publish was getting triggered on one and on_message_acked was getting triggered on other.

**Comment by drasko** [comment]
@abhidtu what's the status on this? Has it been tested in a cluster?

**Comment by Gilbert-Wong** [comment]
So sorry for late response.

1. There is no chance that both A & B may trigger `on_message_publish` hook in their individual plugins.
2. Yes.
